test_partial_write_non_contiguous is flaky on Mac

Should be closed before #759 

May relate to other Mac flaky tests with DuplicateKeyException.

Example:

https://github.com/man-group/ArcticDB/actions/runs/6353734190/job/17258947250

```
 =================================== FAILURES ===================================
___________ test_partial_write_non_contiguous[2-9-cols615-bounds615] ___________

version_store_factory = <function version_store_factory.<locals>.create_version_store at 0x14b7ec4c0>
colnum = 2, rownum = 9, cols = (2, 5, 7), bounds = (16, 27)

    @pytest.mark.parametrize("colnum,rownum,cols,bounds", gen_params_non_contiguous())
    def test_partial_write_non_contiguous(version_store_factory, colnum, rownum, cols, bounds):
        version_store = version_store_factory(col_per_group=colnum, row_per_segment=rownum)
        idx = np.arange(0, 10)
        data = {
            "x": np.arange(10, 20, dtype=np.int64),
            "y": np.arange(20, 30, dtype=np.int64),
            "z": np.arange(30, 40, dtype=np.int64),
        }
        df_orig = pd.DataFrame(data=data, index=idx)
        df = df_orig.pivot(index="y", values="x", columns="z")
    
        sid = "XXX"
>       version_store.write(sid, df)

tests/unit/arcticdb/version_store/test_engine.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = NativeVersionStore: Library: local.test.431_2023-09-29T17_49_41_177185, Primary Storage: lmdb_storage.
symbol = 'XXX'
data = z     30    31    32    33    34    35    36    37    38    39
y                                                      ...aN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  18.0   NaN
29   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  19.0
metadata = None, prune_previous_version = False, pickle_on_failure = False
validate_index = False, kwargs = {}
proto_cfg = column_group_size: 2
segment_row_size: 9
, dynamic_strings = False
recursive_normalizers = False, parallel = False, incomplete = False
coerce_columns = None, sparsify_floats = False

    def write(
        self,
        symbol: str,
        data: Any,
        metadata: Optional[Any] = None,
        prune_previous_version: Optional[bool] = None,
        pickle_on_failure: Optional[bool] = None,
        validate_index: bool = False,
        **kwargs,
    ) -> Optional[VersionedItem]:
        """
        Write `data` to the specified `symbol`. If `symbol` already exists then a new version will be created to
        reference the newly written data. For more information on versions see the documentation for the `read`
        primitive.
    
        Pandas DataFrames, Pandas Series and Numpy NDArrays will be normalised into a common structure suitable for
        storage. Data that cannot be normalised can be written by pickling the data, however pickled data
        consumes more storage space, is less performant for reads and writes and does not support advanced query
        features. Pickling is therefore only supported via the `pickle_on_failure` flag.
    
        Normalised data will be divided into segments that are deduplicated against storage prior to write. As a result,
        if `data` contains only slight changes compared to pre-existing versions only the delta will be written.
    
        Note that `write` is not designed for multiple concurrent writers over a single symbol.
    
        Note: ArcticDB will use the 0-th level index of the Pandas DataFrame for its on-disk index.
    
        Any non-`DatetimeIndex` will converted into an internal `RowCount` index. That is, ArcticDB will assign each
        row a monotonically increasing integer identifier and that will be used for the index.
    
        Parameters
        ----------
        symbol : `str`
            Symbol name. Limited to 255 characters. The following characters are not supported in symbols:
            "*", "&", "<", ">"
        data : `Union[pd.DataFrame, pd.Series, np.array]`
            Data to be written.
        metadata : `Optional[Any]`, default=None
            Optional metadata to persist along with the symbol.
        prune_previous_version : `bool`, default=True
            Removes previous (non-snapshotted) versions from the database.
        pickle_on_failure: `bool`, default=False
            Pickle `data` if it can't be normalized.
        validate_index: bool, default=False
            If True, will verify that the index of `data` supports date range searches and update operations. This in effect tests that the data is sorted in ascending order.
            ArcticDB relies on Pandas to detect if data is sorted - you can call DataFrame.index.is_monotonic_increasing on your input DataFrame to see if Pandas believes the
            data to be sorted
        kwargs :
            passed through to the write handler
    
        Returns
        -------
        Optional[VersionedItem]
            Structure containing metadata and version number of the written symbol in the store.
            The data attribute will not be populated.
    
        Raises
        ------
        UnsortedDataException
            If data is unsorted, when validate_index is set to True.
    
        Examples
        --------
    
        >>> df = pd.DataFrame({'column': [5,6,7]})
        >>> lib.write("symbol", df, metadata={'my_dictionary': 'is_great'})
        >>> lib.read("symbol").data
           column
        0       5
        1       6
        2       7
        """
        self.check_symbol_validity(symbol)
        proto_cfg = self._lib_cfg.lib_desc.version.write_options
    
        dynamic_strings = self._resolve_dynamic_strings(kwargs)
    
        pickle_on_failure = self.resolve_defaults(
            "pickle_on_failure", proto_cfg, global_default=False, existing_value=pickle_on_failure, **kwargs
        )
        prune_previous_version = self.resolve_defaults(
            "prune_previous_version", proto_cfg, global_default=False, existing_value=prune_previous_version, **kwargs
        )
        recursive_normalizers = self.resolve_defaults(
            "recursive_normalizers", proto_cfg, global_default=False, uppercase=False, **kwargs
        )
        parallel = self.resolve_defaults("parallel", proto_cfg, global_default=False, uppercase=False, **kwargs)
        incomplete = self.resolve_defaults("incomplete", proto_cfg, global_default=False, uppercase=False, **kwargs)
    
        # TODO remove me when dynamic strings is the default everywhere
        if parallel:
            dynamic_strings = True
    
        coerce_columns = kwargs.get("coerce_columns", None)
        sparsify_floats = kwargs.get("sparsify_floats", False)
    
        _handle_categorical_columns(symbol, data, False)
    
        log.debug(
            "Writing with pickle_on_failure={}, prune_previous_version={}, recursive_normalizers={}",
            pickle_on_failure,
            prune_previous_version,
            recursive_normalizers,
        )
    
        # Do a multi_key write if the structured is nested and is not trivially normalizable via msgpack.
        if recursive_normalizers:
            vit = self._try_flatten_and_write_composite_object(
                symbol, data, metadata, pickle_on_failure, dynamic_strings, prune_previous_version
            )
            if isinstance(vit, VersionedItem):
                return vit
    
        udm, item, norm_meta = self._try_normalize(
            symbol, data, metadata, pickle_on_failure, dynamic_strings, coerce_columns
        )
        # TODO: allow_sparse for write_parallel / recursive normalizers as well.
        if isinstance(item, NPDDataFrame):
            if parallel:
                self.version_store.write_parallel(symbol, item, norm_meta, udm)
                return None
            elif incomplete:
                self.version_store.append_incomplete(symbol, item, norm_meta, udm)
                return None
            else:
>               vit = self.version_store.write_versioned_dataframe(
                    symbol, item, norm_meta, udm, prune_previous_version, sparsify_floats, validate_index
                )
E               arcticdb_ext.storage.DuplicateKeyException: d:XXX:0:0xaebd0ca463a243e9@1696009781188246000[9,10]
```

--- a/python/tests/unit/arcticdb/version_store/test_engine.py
+++ b/python/tests/unit/arcticdb/version_store/test_engine.py
@@ -12,6 +12,7 @@ from numpy import testing as nt
 import pandas as pd
 import pytest
 import random
+import sys
 from arcticdb.version_store import TimeFrame
 
 
@@ -60,6 +61,7 @@ def test_partial_write_read(version_store_factory, colnum, rownum, cols, tsbound
     assert rtf == vit.data
 
 
+@pytest.mark.skipif(sys.platform == "darwin", reason="Test broken on MacOS (issue #923)")
 @pytest.mark.parametrize("colnum,rownum,cols,bounds", gen_params_non_contiguous())
 def test_partial_write_non_contiguous(version_store_factory, colnum, rownum, cols, bounds):
     version_store = version_store_factory(col_per_group=colnum, row_per_segment=rownum)

